tests/unit/amo/components/TestUserProfilePictureEdit.js

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+
+import UserProfileEditPicture, {
+  UserProfileEditPictureBase,
+} from 'amo/components/UserProfileEditPicture';
+import { getCurrentUser } from 'amo/reducers/users';
+import UserAvatar from 'ui/components/UserAvatar';
+import {
+  dispatchSignInActions,
+} from 'tests/unit/amo/helpers';
+import {
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  const render = ({ i18n = fakeI18n(), ...props } = {}) => {
+    return shallowUntilTarget(
+      <UserProfileEditPicture i18n={i18n} name="input-name" {...props} />,
+      UserProfileEditPictureBase,
+    );
+  };
+
+  it('renders itself', () => {
+    const name = 'some-input-name';
+    const root = render({ name });
+
+    expect(root.find('.UserProfileEditPicture')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit--label'))
+      .toHaveText('Profile photo');
+    expect(root.find('.UserProfileEditPicture-file')).toHaveLength(1);
+    expect(root.find('.UserProfileEditPicture-file-input'))
+      .toHaveProp('name', name);
+    expect(root.find('.UserProfileEditPicture-delete-button')).toHaveLength(0);
+  });
+
+  it('renders a UserAvatar component', () => {
+    const root = render();
+
+    expect(root.find(UserAvatar)).toHaveLength(1);
+    expect(root.find(UserAvatar)).toHaveProp('altText', null);
+  });
+
+  it('specifies the alt text of the UserAvatar component when a user is passed', () => {
+    const { state } = dispatchSignInActions();
+    const user = getCurrentUser(state.users);
+
+    const root = render({ user });
+
+    expect(root.find(UserAvatar))
+      .toHaveProp('altText', `Profile picture for ${user.name}`);
```

Comment:
We might as well also test that the `UserAvatar` component received the expected `preview` and `user` props as well.